docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+
+#### Collect and Push Architecture
+
+Metric data will be collected and cached by each service.  At designated times (kicked off by configurable schedule), the service will collect telemetry data from the cache and push it to a designated message bus topic.
+
+#### Metrics Messaging
+
+Cached metric data, at the designated time, will be marshaled into a message and pushed to the pre-configured message bus topic.
+
+Each metric message consists of several key/value pairs:
+- a **required** name (the name of the metric) such as service-uptime
+- a **required** value which is the telemetry value collected such as 120 as the number of hours the service has been up.
+- a **required** timestamp is the time (in Epoch timestamp/milliseconds format) at which the data was collected (similar in nature to the origin of sensed data). 
+- an optional collection (array) of tags.  The tags are sets of key/value pairs of strings that provide amplifying information about the telemetry.  Tags may include:
+    - originating service name
+    - unit of measure associated with the telemetry value
+    - value type of the value
+    - additional values when the metric is more than just one value (example: when using a histogram, it would include min, max, mean and sum values)
+
+The metric name must be unique for that service.  Because some metrics are reported from multiple services (such as service uptime), the name is not required to be unique across all services.
+
+All information (keys, values, tags, etc.) is in string format and placed in a JSON array within the message body.  Here are some example representations:
+
+**Example metric message body with a single value**
+{"name":"service-up", "value":"120", "timestamp":"1602168089665570000", "tags":{"service":"coredata","uom":"days","type":"int64"}}
+
+**Example metric message body with multiple values**
+{"name":"api-requests", "value":"24", "timestamp":"1602168089665570001", "tags":{"service":"coredata","uom":"count","type":"int64", "mean":"0.0665", "rate1":"0.111", "rate5":"0.150","rate15":"0.111"}}
+
+!!! Note
+    The key or metric name must be unique when using go-metrics as it requires the metric name to be unique per the registry.  Metrics are considered immutable.
+
+#### REST endpoints
+
+!!! Note
+    Again, these REST endpoints are per service and meant to control which metrics for a service are turned on or off.  The REST endpoints do not provide metrics data.
+
+- Proposed endpoint for the REST endpoint that responds with what metrics the service offers is:  /api/v2/telemetry
+- Body of the GET response would contain a JSON list of metrics (by metric name key) and an indication of which are `on` and which are `off`
+- Body of the PUT request would contain a JSON list of the metrics that are to be turned `on` (others are assumed to be turned `off`)
+- Body of the PATCH request would contain a JSON list of the metrics that are to be turned `on` or `off` - leaving all other metrics unchanged

```

Comment:
There's no mention of scheduling?

Reply:
@tonyespy , see `interval` below which is not part of this API.

Reply:
Given there is no API left in the service to have the schedule service to call, we have to do this internally now.  I've suggested that we do the scheduling like autoevent does it in device services.